src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -501,6 +501,10 @@ private void innerCreateNoLock(Create create, IndexWriter writer, long currentVe
                 // #7142: the primary already determined it's OK to index this document, and we confirmed above that the version doesn't
                 // conflict, so we must also update here on the replica to remain consistent:
                 doUpdate = true;
+            } else if (create.origin() == Operation.Origin.PRIMARY && create.autoGeneratedId() && create.canHaveDuplicates() && currentVersion == 1 && create.version() == Versions.MATCH_ANY) {
+                doUpdate = true;
+                updatedVersion = 1;
+
```

Comment:
extra newline.... I think this optimization deserves a rather longish comment. can you add it?